src/internal/test_utils.rs

```diff
@@ -0,0 +1,165 @@
+//! Testing utilities.
+
+use std::sync::LazyLock;
+
+use crate::common::{Graph, Nodes};
+
+pub mod exports {
+    pub use std::cmp;
+
+    pub use hashbrown::{HashMap, HashSet};
+}
+
+macro_rules! nodes {
+    ($($x:expr),*) => {
+        $crate::internal::test_utils::exports::HashSet::from_iter([$($x),*].iter().copied())
+    };
+}
+
+macro_rules! measurements {
+    ($($u:literal: $v:expr),*) => {
+        $crate::internal::test_utils::exports::HashMap::from_iter([$(($u, ($v).into())),*].iter().copied())
+    };
+}
+
+macro_rules! static_max {
+    ($x:expr) => {
+        $x
+    };
+    ($x:expr, $($y:expr),+) => {
+        $x.max(static_max!($($y),+))
+    };
+}
+
+macro_rules! graph {
+    () => {
+        vec![]
+    };
+    ($(($u:expr, $v:expr)),+) => {{
+        let n = $crate::internal::test_utils::exports::cmp::max(static_max!($($u),+), static_max!($($v),+)) + 1;
+        let mut g = vec![$crate::internal::test_utils::exports::HashSet::new(); n];
+        $(
+        g[$u].insert($v);
+        g[$v].insert($u);
+        )+
+        g
+    }};
+}
```

Comment:
`graph` could be define as a usual function: 
```suggestion
fn graph<const N: usize>(adj: &[(usize, usize); N]) -> Graph {
    match adj.iter().map(|(u, v)| [u, v].into_iter()).flatten().max() {
        None => vec![],
        Some(n) => {
            let mut g = vec![Nodes::new(); n + 1];
            for &(u, v) in adj {
                g[u].insert(v);
                g[v].insert(u);
            };
            g
        }
    }
}
```

Reply:
Merged in 77a4602526fa0d77420ca72bcca44c757d5113ce.